### PR TITLE
Add deployment config and docu.

### PR DESCRIPTION
**Pull request by clemensmanert**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,3 +94,93 @@ python -m unittest
...
 pylint app.py apitests models routes tests util
 ```
+## How to deploy
+This chapter describes how to get a production instance running.
+### Setup a new instance
+Clone the sources:
+```bash
+git clone https://github.com/buzzn/mybuzzn-backend.git
+```
+We recommend to setup a virtual env for the projects' requirements.
```
Comment (missgreenwood) [inline README.md]: Maybe replace "virtual env" by "virtual environment" to be clear.

File: README.md [hunk 2]
```diff
@@ -94,3 +94,93 @@ python -m unittest
...
+cd PROJECT_ROOT
+virtualenv -p `whereis python | cut -d ' ' -f3` .venv
+source .venv/bin/activate
+```
+Install the requirements
+```bash
+pip install -r "requirements.txt"
+pip install git+https://github.com/buzzn/discovergy.git
+```
+Create an endpoint in your webserver, for example for apache, create a new site:
```
Comment (missgreenwood) [inline README.md]: "Apache"

File: README.md [hunk 3]
```diff
@@ -94,3 +94,93 @@ python -m unittest
...
+```
+<VirtualHost *:80>
+  ServerName your-domain.net
+  ErrorLog /path/to/error/log
+  CustomLog /path/to/access.log combined
+
+  WSGIDaemonProcess mybuzznbackend user=mybuzznbackend-user group=mybuzznbackend-group threads=5 python-path=/path/to/python:/and/path/to/packages
+  WSGIScriptAlias / /path/to/mybuzzn/projectdir/mybuzznbackend.wsgi
+  WSGIPassAuthorization On # Importent, so autho header is passed to the app
```
Comment (missgreenwood) [inline README.md]: "important for authorization header to be passed to the app"

File: README.md [hunk 4]
```diff
@@ -94,3 +94,93 @@ python -m unittest
...
+
+    <Directory /path/to/project/>
+        WSGIProcessGroup mybuzznbackend
+        WSGIApplicationGroup %{GLOBAL}
+        Order deny,allow
+        Allow from all
+    </Directory>
+</VirtualHost>
+```
+Make sure that `WSGIPassAuthorization` is set to `On`, to have the app access to
```
Comment (missgreenwood) [inline README.md]: "Make sure that `WSGIPassAuthorization` is set to `On` to grant the app authoriziation header access."

File: README.md [hunk 5]
```diff
@@ -94,3 +94,93 @@ python -m unittest
...
+the auth headers.
+
+Make sure a file `setup_environment.py` exists in the project root, which sets
```
Comment (missgreenwood) [inline README.md]: "Make sure a file `setup_environment.py` exists in the project root which sets environment variables like this:"

File: README.md [hunk 6]
```diff
@@ -94,3 +94,93 @@ python -m unittest
...
+User=mybuzznbackend-user
+Group=mybuzznbackend-group
+
+[Install]
+WantedBy=multi-user.target⏎
+```
+
+### Upgrading to a new version
+To upgrade to a new version, go to the project root and run `git pull`.
+If something has changed on the models, run `source ./env/bin/activate` to
```
Comment (missgreenwood) [inline README.md]: "If something has changed in the database models run `source ./.venv/bin/activate`..." to be consistent with the naming from above.

File: README.md [hunk 7]
```diff
@@ -94,3 +94,93 @@ python -m unittest
...
+activate the python environment. Then run `./env/bin/flask db upgrade` to
```
Comment (missgreenwood) [inline README.md]: Same here: "Then run `./.venv/bin/flask db upgrade` ..."

File: README.md [hunk 8]
```diff
@@ -94,3 +94,93 @@ python -m unittest
...
+upgrade the database.
+
+Finally restart the redis task and the webserver:
```
Comment (missgreenwood) [inline README.md]: "the redis service"

File: mybuzznbackend.wsgi [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import setup_environment
+
+from app import app as application
```
Comment (missgreenwood) [inline mybuzznbackend.wsgi]: Maybe remove the strangely encoded newline at the end of this line?

**Review by missgreenwood — APPROVED** [review]
I like it a lot! Please consider the comments I made, mainly regarding the language.

File: README.md (hunk 8 above)
Comment (clemensmanert) [reply README.md]: No, i meant the tasks which fills the redis keyvaluestore with meter readings. I improved that sentence :)

**Pull request merged**